src/vos/tests/vts_ts.c

```diff
@@ -402,6 +415,154 @@ lru_array_test(void **state)
 	assert_false(found);
 }
 
+#define STRESS_ITER 500
+#define BIG_TEST 50000
+static void
+lru_array_stress_test(void **state)
+{
+	struct lru_arg		*ts_arg = *state;
+	struct lru_record	*entry;
+	struct index_record	*stress_entries;
+	int			 evicted, inserted;
+	int			 i, op, j;
+	bool			 found;
+	int			 freq_map[] = {2, 3, 7, 13, 17};
+	int			 freq_idx;
+	int			 freq_idx2;
+	int			 freq;
+
+	D_ALLOC_ARRAY(stress_entries, BIG_TEST);
+	assert_non_null(stress_entries);
+
+	for (j = 0; j < STRESS_ITER * ARRAY_SIZE(freq_map); j++) {
+		freq_idx = j % ARRAY_SIZE(freq_map);
+		/** First evict all */
+		for (i = 0; i < NUM_INDEXES; i++) {
+			found = lrua_lookup(ts_arg->array,
+					    &ts_arg->indexes[i].idx,
+					    (void **)&entry);
+			assert_false(found);
+		}
+		/** Now insert most */
+		for (i = 0; i < NUM_INDEXES; i++) {
+			if ((i % freq_map[freq_idx]) == 0)
+				continue;
+			entry = lrua_alloc(ts_arg->array,
+					   &ts_arg->indexes[i].idx, true);
+			assert_non_null(entry);
+			entry->record = &ts_arg->indexes[i];
+			ts_arg->indexes[i].value = i;
+		}
+		freq_idx2 = (freq_idx + 1) % ARRAY_SIZE(freq_map);
+		freq = freq_map[freq_idx] * freq_map[freq_idx2];
+		/** Now evict some of them */
+		evicted = 0;
+		for (i = NUM_INDEXES - 1; i >= 0; i--) {
+			if ((i % freq) == 0)
+				continue;
+			found = lrua_lookup(ts_arg->array,
+					    &ts_arg->indexes[i].idx,
+					    (void **)&entry);
+			if (!found)
+				continue;
+
+			assert_non_null(entry);
+			assert_true(entry->magic1 == MAGIC1);
+			assert_true(entry->magic2 == MAGIC2);
+
+			evicted++;
+
+			lrua_evict(ts_arg->array, &ts_arg->indexes[i].idx);
+		}
+
+		assert_int_equal(evicted, LRU_ARRAY_SIZE);
```

Comment:
Would you please to comment why "evicted" must be equal to LRU_ARRAY_SIZE?

Reply:
done